File: app/src/main/java/com/example/kotlin/java/MyEnum2.java
package com.example.kotlin.java;

import android.text.TextUtils;

/**
 * @author zhangquan
 */
public enum MyEnum2 {
    ONE("ONE"),
    TWO("TWO");

    public String name;

    private MyEnum2(String name) {
        this.name = name;
    }

    public static MyEnum2 getByName(String name) {
        MyEnum2[] values = MyEnum2.values();
        for (MyEnum2 item : values) {
            if (TextUtils.equals(item.name, name)) {
                return item;
            }
        }
        return null;
    }
}

File: app/src/main/java/com/example/kotlin/kotlin/enums/MyEnum2.kt
package com.example.kotlin.kotlin.enums

import android.text.TextUtils

/**
 * @author zhangquan
 */
enum class MyEnum2(name: String) { //带构造函数的枚举
    ONE("ONE"), TWO("TWO");

    companion object {
        fun getByName(name: String?): MyEnum2? {
            val values =values()
            for (item in values) {
                if (TextUtils.equals(item.name, name)) {
                    return item
                }
            }
            return null
        }
    }

}